retroshare-service/src/retroshare-service.cc

```diff
@@ -17,66 +17,274 @@
  */
 
 #include "util/stacktrace.h"
+#include "util/argstream.h"
+#include "retroshare/rsinit.h"
+#include "jsonapi/jsonapi.h"
 
 CrashStackTrace gCrashStackTrace;
 
-#include <QCoreApplication>
+#include <cmath>
 #include <csignal>
-#include <QObject>
-#include <QStringList>
+
+#ifndef __ANDROID__
+#include <termios.h>
+#endif
 
 #ifdef __ANDROID__
 #	include <QAndroidService>
+#	include <QCoreApplication>
+#	include <QObject>
+#	include <QStringList>
+
+#	include "util/androiddebug.h"
 #endif // def __ANDROID__
 
 #include "retroshare/rsinit.h"
 #include "retroshare/rsiface.h"
 
-#ifdef __ANDROID__
-#	include "util/androiddebug.h"
-#endif
-
 #ifndef RS_JSONAPI
 #	error Inconsistent build configuration retroshare_service needs rs_jsonapi
 #endif
 
+int getch() {
+    int ch;
+    struct termios t_old, t_new;
+
+    tcgetattr(STDIN_FILENO, &t_old);
+    t_new = t_old;
+    t_new.c_lflag &= ~(ICANON | ECHO);
+    tcsetattr(STDIN_FILENO, TCSANOW, &t_new);
+
+    ch = getchar();
+
+    tcsetattr(STDIN_FILENO, TCSANOW, &t_old);
+    return ch;
+}
+
+std::string readStringFromKeyboard(const char *prompt, bool show_asterisk=true)
+{
+  const char BACKSPACE=127;
```

Comment:
use `constexpr` here too

Reply:
constexpr is actually not needed for integral expressions. See https://stackoverflow.com/questions/14116003/difference-between-constexpr-and-const